zebrad/src/components/inbound.rs

```diff
@@ -53,52 +99,27 @@ pub type SetupData = (Outbound, Arc<Mutex<AddressBook>>);
 /// responding to block gossip by attempting to download and validate advertised
 /// blocks.
 pub struct Inbound {
-    // invariant: address_book, outbound, downloads are Some if network_setup is None
-    //
-    // why not use an enum for the inbound state? because it would mean
-    // match-wrapping the body of Service::call rather than just expect()ing
-    // some Options.
-
-    // Setup
-    /// A oneshot channel used to receive the address_book and outbound services
-    /// after the network is set up.
-    network_setup: Option<oneshot::Receiver<SetupData>>,
-
-    // Services
-    /// A list of peer addresses.
-    address_book: Option<Arc<Mutex<zn::AddressBook>>>,
-
-    /// A service that downloads and verifies gossipped blocks.
-    downloads: Option<Pin<Box<Downloads<Timeout<Outbound>, Timeout<Verifier>, State>>>>,
-
-    /// A service that forwards requests to connected peers, and returns their
-    /// responses.
+    /// Provides network-dependent services, if they are available.
     ///
-    /// Only used for readiness checks, and via `downloads`.
-    outbound: Option<Outbound>,
+    /// Some services are unavailable until Zebra has completed network setup.
+    network: Setup,
```

Comment:
I think we should line up the variable and type names to be a little more consistent. Right now reading the code below is a little confusing because its using different terminology interchangeably and its hard to keep things lined up mentally. I don't really have a preference on how, some suggestions:

* `network_setup: Setup`
* `network: NetworkState`
* `network: Network`
* `setup: Setup`

Reply:
I went with `network_setup: Setup`